composer.json

```diff
@@ -14,13 +14,24 @@
     }
   ],
   "autoload": {
-    "psr-0": {
-      "PAMI": "src/mg"
+    "psr-4": {
+      "PAMI\\": "src/mg/PAMI"
     }
   },
   "require": {
     "php": ">=5.3.3",
     "apache/log4php": "2.3.0",
     "psr/log": "~1.0"
+  },
+  "require-dev": {
+    "phpunit/phpunit": "4.8.*",
+    "phing/phing": "2.*",
+    "squizlabs/php_codesniffer": "2.*",
+    "sebastian/phpcpd": "*",
```

Comment:
Well, it may be my cautious nature to always specify a version constraint.
But if you only use them as executables, and not as a library (even in the
tests) then I think it will be fine, indeed.

On Thu, 20 Aug 2015 at 23:25 Marcelo Gornstein <notifications@github.com>
wrote:

> In composer.json
> <https://github.com/marcelog/PAMI/pull/84#discussion_r37583091>:
>
> >      }
> >    },
> >    "require": {
> >      "php": ">=5.3.3",
> >      "apache/log4php": "2.3.0",
> >      "psr/log": "~1.0"
> > +  },
> > +  "require-dev": {
> > +    "phpunit/phpunit": "4.8.*",
> > +    "phing/phing": "2.*",
> > +    "squizlabs/php_codesniffer": "2.*",
> > +    "sebastian/phpcpd": "*",
>
> I'd agree with regular depencies.. but in this case these tools are not
> used *programatically* from the code but *run* to analyse the code
> itself, so we *should* be ok (they are also only required for the dev
> environment). The idea is to actually be updated to the latest version in
> case the new versions are faster or better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/marcelog/PAMI/pull/84/files#r37583091>.
>
-- 
Met vriendelijke groet,

Jacob Kiers